### PR TITLE
chore: add segment user cache processor

### PR DESCRIPTION
**Pull request by cre8ivejp**
Part of https://github.com/bucketeer-io/bucketeer/issues/921

### CODE REVIEW

File: pkg/bucketeer/cache/processor/segment_user_processor.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		requestedAt,
+	)
+	// Get the latest cache from the server
+	resp, size, err := p.apiClient.GetSegmentUsers(req)
+	if err != nil {
+		return err
+	}
+	p.loggers.Debugf("bucketeer/cache: GetSegmentUsers response: %v, size: %d", resp, size)
+	// Delete all the local cache and save the new one
+	if resp.ForceUpdate {
```
Comment (cre8ivejp) [inline pkg/bucketeer/cache/processor/segment_user_processor.go:126]: If the `ForceUpdate` is true, it must delete all the local cache and save the new one from the response.

File: pkg/bucketeer/cache/processor/segment_user_processor.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+		return p.deleteAllAndSaveLocalCache(resp.RequestedAt, resp.SegmentUsers)
+	}
+	// Update only the updated flags
+	return p.updateLocalCache(
```
Comment (cre8ivejp) [inline pkg/bucketeer/cache/processor/segment_user_processor.go:130]: If the `ForceUpdate` is false, the updated and deleted segments will be updated.

File: pkg/bucketeer/cache/processor/segment_user_processor.go [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+}
+
+func (p *segmentUserProcessor) runProcessLoop() {
+	defer func() {
+		p.loggers.Debug("bucketeer/cache: segmentUsers runProcessLoop done")
+	}()
+	ticker := time.NewTicker(p.pollingInterval)
+	defer ticker.Stop()
+	// Update the cache once when starts polling
+	if err := p.updateCache(); err != nil {
```
Comment (cre8ivejp) [inline pkg/bucketeer/cache/processor/segment_user_processor.go:84]: Update the cache once when starting the processor.

**Review by Ubisoft-potato — APPROVED** [review]
LGTM! 🚀

**Pull request merged**